api/src/main/java/ca/bc/gov/educ/api/trax/service/institute/CodeService.java

```diff
@@ -99,4 +136,33 @@ public void loadSchoolFundingGroupCodesIntoRedisCache(List<SchoolFundingGroupCod
 		schoolFundingGroupCodeRedisRepository
 				.saveAll(schoolFundingGroupCodeTransformer.transformToEntity(schoolFundingGroupCodes));
 	}
+
+	public List<SchoolFundingGroupCode> getSchoolFundingGroupCodesFromRedisCache() {
+		log.debug("**** Getting school funding group codes from Redis Cache.");
+		return  schoolFundingGroupCodeTransformer.transformToDTO(schoolFundingGroupCodeRedisRepository.findAll());
+	}
+
+	public void initializeSchoolFundingGroupCodeCache(boolean force) {
+		String cacheStatus = redisTemplate.opsForValue().get(CacheKey.SCHOOL_FUNDING_GROUP_CODE_CACHE.name());
+		cacheStatus = cacheStatus == null ? "" : cacheStatus;
+		if (CacheStatus.LOADING.name().compareToIgnoreCase(cacheStatus) == 0
+				|| CacheStatus.READY.name().compareToIgnoreCase(cacheStatus) == 0) {
+			log.info(String.format("SCHOOL_FUNDING_GROUP_CODE_CACHE status: %s", cacheStatus));
+			if (force) {
+				log.info("Force Flag is true. Reloading SCHOOL_FUNDING_GROUP_CODE_CACHE...");
```

Comment:
Reduce duplicate code block by declaring a method. if(force) and else statement duplicate lines of code.

Reply:
Fixed here: https://github.com/bcgov/EDUC-GRAD-TRAX-API/blob/c79bdcc841ed4529314fd9ef0d3bd821fb053b15/api/src/main/java/ca/bc/gov/educ/api/trax/service/institute/ServiceHelper.java#L24
Added new generic class ServiceHelper used by other institute service classes.